subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractAndroidSantaTrackerSmokeTest.groovy

```diff
@@ -56,6 +66,22 @@ class AbstractAndroidSantaTrackerSmokeTest extends AbstractSmokeTest {
             if (GradleContextualExecuter.notConfigCache) {
                 expectProjectConventionDeprecationWarning(agpVersion)
                 expectAndroidConventionTypeDeprecationWarning(agpVersion)
+                expectConfigUtilDeprecationWarning(agpVersion)
+            }
+        }.build()
+    }
+
+    protected BuildResult buildUpToDateLocation(File projectDir, String agpVersion) {
+        return runnerForLocation(projectDir, agpVersion, "assembleDebug").deprecations(SantaTrackerDeprecations) {
+            if (GradleContextualExecuter.notConfigCache) {
+                expectProjectConventionDeprecationWarning(agpVersion)
+                expectAndroidConventionTypeDeprecationWarning(agpVersion)
+                expectConfigUtilDeprecationWarning(agpVersion)
+            } else {
+                // TODO - this is here because AGP >=7.4 reads build/generated/source/kapt/debug at configuration time
```

Comment:
I read that comment as meaning something else than the code. I think the following suggestion clarifies it:
```suggestion
                // TODO - this is here because AGP 7.4.x reads build/generated/source/kapt/debug at configuration time
```